pxtsim/runtime.ts

```diff
@@ -450,13 +473,18 @@ namespace pxsim {
         recordingWidth: number = undefined;
         startTime = 0;
         startTimeUs = 0;
+        pausedTime = 0;
+        lastPauseTimestamp = 0;
         id: string;
         globals: any = {};
         currFrame: StackFrame;
         entry: LabelFn;
         loopLock: Object = null;
         loopLockWaitList: (() => void)[] = [];
 
+        timeoutsScheduled: TimeoutScheduled[] = []
```

Comment:
Why two arrays? Seems like only one will ever be used at a time; you're either paused or running.

Reply:
That's true, only one of them is going to be used at once. But I think having two arrays makes it more clear, since their contents are different. And other issues arise  if we were to use only one, like what would be the ID of a paused timeout? It could be the one from the first call, but it's gonna change when we resume it.

Reply:
I can change it if you disagree with that